include/swift/AST/DiagnosticsSema.def

```diff
@@ -4749,15 +4749,10 @@ ERROR(could_not_infer_pack_element,none,
 NOTE(note_in_opening_pack_element,none,
       "in inferring pack element #%0 of '%1'", (unsigned,StringRef))
 
-      
-ERROR(type_of_expression_is_ambiguous,none,
-      "type of expression is ambiguous without a type annotation", ())
-
 ERROR(failed_to_produce_diagnostic,Fatal,
```

Comment:
Not actually sure if we want to keep this as a fatal error given we'll now be emitting it in more cases, thoughts?